core/src/main/java/org/apache/accumulo/core/security/crypto/CryptoModuleFactory.java

```diff
@@ -96,23 +95,15 @@ private static CryptoModule instantiateCryptoModule(String cryptoModuleClassname
     }
 
     if (!implementsCryptoModule) {
-      log.warn("Configured Accumulo crypto module \"{}\" does not implement the CryptoModule interface. No encryption will be used.", cryptoModuleClassname);
-      return new NullCryptoModule();
+      throw new IllegalArgumentException("Configured Accumulo crypto module " + cryptoModuleClassname + " does not implement the CryptoModule interface.");
```

Comment:
I'm not sure any of these checks in this section are necessary if the class was properly loaded with the ConfigurationTypeHelper from the configuration. It may just be redundant.

Reply:
For this and the SecretKeyEncryptionStrategy, they might be redundant. I am not sure if anything can or would change these values both after the configuration is read and before the class is loaded here. Leaving them in for now is safer.

Reply:
Agreed. This can be refactored/deferred to later. It's safer to leave it as is for now.